server/src/main/java/org/opensearch/search/query/QueryPhase.java

```diff
@@ -131,13 +131,15 @@ public void preProcess(SearchContext context) {
     }
 
     public void execute(SearchContext searchContext) throws QueryPhaseExecutionException {
```

Comment:
Hi @peteralfonsi, could you please evaluate the use of [Timer](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/profile/Timer.java) class for this change? I'd like to hear your thoughts on whether utilizing the existing Timer class is a feasible choice. Thank you.